src/sass/abstractions/icons/_icon-mixin.scss

```diff
@@ -0,0 +1,108 @@
+@import 'init';
+
+@import '../../dist/svg/svg-sprite-mixins';
+@import '../../dist/fonts/svg-font-mixins';
+
+// g-icon()
+//
+// Global mixin for include colored icon.
+// With modernizr check if browser supports svg-fragment and css-mask
+// then use: svg-sprite, otherwice generated iconfont.
+//
+// Should be added to any component and will add icon on pseudo :: before.
+//
+// $name = Filename of svg. Maybe located inside
+// options.theme.sass + '/storage/svg/*.svg'
+// See `iconfont` and `svg-sprites` gulp tasks.
+//
+// $color: functional or base color name from _colors.scss
+//
+// $fallback-font-size: font-size of fallback in case of iconfont
+// possible values:
+//                  `NUMpx` only.
+//
+// $display: set value of `display` property for icon in case of svg-sprite.
+// possible values:
+//                  `block` (default)
+//                  `inline-block`
+//
+// $where-to-insert: in what placeholder icon should be added
+// possible values:
+//                  `before` (default, if icon should be added before content, or not necessary)
+//                  `after` (if icon should be added after content)
+
+@mixin g-icon($name, $color: brand_1, $fallback-font-size: inherit, $display: block, $where-to-insert: before) {
+  .cssmask & {
+    @if ($where-to-insert == before) {
+      &::before {
```

Comment:
it can look prettier with 
``` 
&::#{$where-to-insert} {}
```
without if statement

Reply:
done